plugin/src/main/scala/org/virtuslab/akkasaferserializer/SaferSerializerPluginComponent.scala

```diff
@@ -0,0 +1,64 @@
+package org.virtuslab.akkasaferserializer
+
+import scala.tools.nsc.plugins.PluginComponent
+import scala.tools.nsc.{Global, Phase}
+
+class SaferSerializerPluginComponent(val pluginOptions: PluginOptions, val global: Global) extends PluginComponent {
+  import global._
+  override val phaseName: String = "akka-safer-serializer-gather"
+  override val runsAfter: List[String] = List("refchecks")
+
+  var annotatedTraitsCache: List[Type] = List()
+
+  override def newPhase(prev: Phase): Phase =
+    new StdPhase(prev) {
+      override def apply(unit: global.CompilationUnit): Unit = {
+        val body = unit.body
+        val reporter = CrossVersionReporter(global)
+
+        val genericsNames = Seq(
+          "akka.actor.typed.Behavior",
+          "akka.persistence.typed.scaladsl.ReplyEffect",
```

Comment:
Pls add `Effect` here as well

Reply:
And open an issue for adding more types (I suppose we should check a few dozen different types to make this plugin really production-grade)

Reply:
> Pls add `Effect` here as well

Could you provide a FQCN of `Effect`?

Reply:
Sorry, I meant `akka.persistence.typed.scaladsl.Effect`